src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,16 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    int x = std::min(0, mBounds.x());
+    int y = std::min(0, mBounds.y());
+    int width = std::max(mWidth, mBounds.width());
+    int height = std::max(mHeight, mBounds.height());
```

Comment:
Please remove this and document where it is causing issues, so that we can find a way to fix that.